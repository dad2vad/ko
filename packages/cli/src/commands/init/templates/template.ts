import { readFile } from "node:fs/promises";
import { join } from "node:path";
import { fileURLToPath } from "node:url";
import { File } from "./base";

const TEMPLATE_DIR = join(
  fileURLToPath(import.meta.url),
  // eslint-disable-next-line no-restricted-globals
  process.env.BUILD_PROD ? "../../templates" : "../../../../../templates"
);

export async function generateFromTemplateFile(
  path: string,
  template = path
): Promise<File> {
  return {
    path,
    content: await readFile(join(TEMPLATE_DIR, template), "utf8")
  };
}

export function generateReadme() {
  return generateFromTemplateFile("README.md");
}
