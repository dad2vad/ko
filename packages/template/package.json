{
  "name": "@kosko/template",
  "version": "3.0.1",
  "description": "Utilities for kosko templates.",
  "homepage": "https://kosko.dev",
  "repository": "https://github.com/tommy351/kosko/tree/master/packages/template",
  "author": "Tommy Chen <tommy351@gmail.com>",
  "license": "MIT",
  "main": "dist/index.node.cjs",
  "module": "dist/index.node.mjs",
  "types": "dist/types.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "build-scripts"
  },
  "exports": {
    "types": "./dist/index.d.ts",
    "deno": "./dist/index.deno.mjs",
    "node": {
      "import": "./dist/index.node.mjs",
      "require": "./dist/index.node.cjs"
    },
    "import": "./dist/index.base.mjs",
    "require": "./dist/index.node.cjs"
  },
  "files": [
    "/dist/"
  ],
  "keywords": [
    "kosko"
  ],
  "publishConfig": {
    "access": "public"
  },
  "engines": {
    "node": ">=14.18.0"
  },
  "dependencies": {
    "@kosko/log": "workspace:^",
    "exit": "^0.1.2",
    "yargs": "^17.5.1"
  },
  "devDependencies": {
    "@kosko/build-scripts": "workspace:^",
    "@kosko/jest-preset": "workspace:^",
    "@kosko/test-utils": "workspace:^",
    "@types/exit": "^0.1.31",
    "@types/yargs": "^17.0.12",
    "execa": "^5.1.1"
  },
  "jest": {
    "preset": "@kosko/jest-preset"
  }
}
