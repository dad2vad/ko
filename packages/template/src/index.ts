/**
 * Template utilities.
 *
 * @packageDocumentation
 */

export * from "./cli";
export * from "./template";
export * from "./write";
