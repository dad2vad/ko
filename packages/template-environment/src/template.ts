import { Template } from "@kosko/template";
import { join } from "node:path";

/**
 * @public
 */
export interface Options {
  name: string;
  esm: boolean;
}

const esmContent = `export default {}`;

const cjsContent = `"use strict";

module.exports = {};
`;

/**
 * @public
 */
export const template: Template<Options> = {
  description: "Create a new environment",
  options: {
    name: {
      type: "string",
      description: "Environment name",
      required: true
    },
    esm: {
      type: "boolean",
      description: "Generate ECMAScript module (ESM) files",
      // eslint-disable-next-line no-restricted-globals
      default: process.env.BUILD_TARGET !== "node"
    }
  },
  async generate({ name, esm }) {
    return {
      files: [
        {
          path: join("environments", name, "index.js"),
          content: esm ? esmContent : cjsContent
        }
      ]
    };
  }
};
