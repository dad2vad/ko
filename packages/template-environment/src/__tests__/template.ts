import { template, Options } from "../template";

async function generate(options: Partial<Options> = {}) {
  const result = await template.generate({
    name: "dev",
    esm: false,
    ...options
  });

  return result.files.map((file) => ({
    ...file,
    path: file.path.replace(/\\/g, "/")
  }));
}

describe("when esm = false", () => {
  test("should return files in CJS format", async () => {
    await expect(generate({ esm: false })).resolves.toMatchSnapshot();
  });
});

describe("when esm = true", () => {
  test("should return files in ESM format", async () => {
    await expect(generate({ esm: true })).resolves.toMatchSnapshot();
  });
});
