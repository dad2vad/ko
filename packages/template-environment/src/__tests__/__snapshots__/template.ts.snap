// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when esm = false should return files in CJS format 1`] = `
[
  {
    "content": ""use strict";

module.exports = {};
",
    "path": "environments/dev/index.js",
  },
]
`;

exports[`when esm = true should return files in ESM format 1`] = `
[
  {
    "content": "export default {}",
    "path": "environments/dev/index.js",
  },
]
`;
